v2/datastream-to-bigquery/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToBigQueryIT.java

```diff
@@ -0,0 +1,209 @@
+import org.apache.beam.it.common.PipelineLauncher.LaunchConfig;
+import org.apache.beam.it.common.PipelineOperator;
+import org.apache.beam.it.common.utils.PipelineUtils;
+import org.apache.beam.it.common.utils.ResourceManagerUtils;
+import org.apache.beam.it.gcp.TemplateTestBase;
+import org.apache.beam.it.gcp.bigquery.BigQueryResourceManager;
+import org.apache.beam.it.gcp.bigquery.conditions.BigQueryRowsCheck;
+import org.apache.beam.it.gcp.datastream.DatastreamResourceManager;
+import org.apache.beam.it.gcp.datastream.MySQLSource;
+import org.apache.beam.it.gcp.pubsub.PubsubResourceManager;
+import org.apache.beam.it.jdbc.CustomMySQLResourceManager;
+import org.apache.beam.it.jdbc.JDBCResourceManager;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.testcontainers.shaded.org.apache.commons.lang3.RandomStringUtils;
+
+@TemplateIntegrationTest(DataStreamToBigQuery.class)
+@RunWith(JUnit4.class)
+public class DataStreamToBigQueryIT extends TemplateTestBase {
+  private CustomMySQLResourceManager jdbcResourceManager;
+  private PubsubResourceManager pubsubResourceManager;
+  private DatastreamResourceManager datastreamResourceManager;
+  private BigQueryResourceManager bigQueryResourceManager;
+
+  private static final int ROW_COUNT = 10;
+
+  private static final String ID = "id";
+  private static final String JOB = "job";
+  private static final String NAME = "name";
+
+  @Before
+  public void setUp() throws IOException {
+    pubsubResourceManager =
+        PubsubResourceManager.builder(testName, PROJECT, credentialsProvider).build();
+    bigQueryResourceManager =
+        BigQueryResourceManager.builder(testName, PROJECT, credentials).build();
+    datastreamResourceManager =
+        DatastreamResourceManager.builder(testName, PROJECT, REGION)
+            .setCredentialsProvider(credentialsProvider)
+            .setPrivateConnectivity("datastream-private-connect-us-central1")
+            .build();
+    jdbcResourceManager = CustomMySQLResourceManager.builder(testName).build();
+  }
+
+  @After
+  public void cleanUp() {
+    ResourceManagerUtils.cleanResources(
+        jdbcResourceManager,
+        pubsubResourceManager,
+        datastreamResourceManager,
+        bigQueryResourceManager);
+  }
+
+  @Ignore("Flaky test")
+  @Test
+  public void testDataStreamToBigQuery() throws IOException {
+    baseDataStreamToBigQuery(Function.identity());
+  }
+
+  @Ignore("Flaky test")
+  @Test
+  public void testDataStreamToBigQueryUsingAtleastOnceMode() throws IOException {
+    ArrayList<String> experiments = new ArrayList<>();
+    experiments.add("streaming_mode_at_least_once");
+    baseDataStreamToBigQuery(
+        b ->
+            b.addEnvironment("additionalExperiments", experiments)
+                .addEnvironment("enableStreamingEngine", true));
+  }
```

Comment:
Is there any sort of check we can do to ensure the atLeastOnce behavior is actually invoked?

Reply:
There are some logs that we can look out for and there is a field in the UI, that denotes that but I don't think there is a surefire way to tell, based on output since that itself is not deterministic, it is At Least Once.

Reply:
Hm ok fair enough. In the future it might be nice to figure out a way to reliably test (even if it is some sort of API call to the job to make sure it was enabled) but not a blocker.